File: dineq-backend/internal/domain/item.go
package domain

import (
	"context"
	"time"
)

type Item struct {
	ID              string           `json:"id"`
	Name            string           `json:"name"`
	NameAm          string           `json:"name_am"`
	Slug            string           `json:"slug"`
	MenuSlug        string           `json:"menu_slug"`
	Description     string           `json:"description"`
	DescriptionAm   string           `json:"description_am"`
	Image           []string         `json:"image"`
	ThumbnailImages []string         `json:"thumbnail_images"`
	Price           float64          `json:"price"`
	Currency        string           `json:"currency"`
	Allergies       []string         `json:"allergies"`
	AllergiesAm     string           `json:"allergies_am"`
	UserImages      []string         `json:"user_images"`
	TabTags         []string         `json:"tab_tags"`
	TabTagsAm       []string         `json:"tab_tags_am"`
	Calories        int              `json:"calories"`
	Protein         int              `json:"protein"`
	Carbs           int              `json:"carbs"`
	Fat             int              `json:"fat"`
	NutritionalInfo *NutritionalInfo `json:"nutritional_info,omitempty"`
	Ingredients     []string         `json:"ingredients"`
	IngredientsAm   []string         `json:"ingredients_am"`
	PreparationTime int              `json:"preparation_time"`
	HowToEat        string           `json:"how_to_eat"`
	HowToEatAm      string           `json:"how_to_eat_am"`
	CreatedAt       time.Time        `json:"created_at"`
	UpdatedAt       time.Time        `json:"updated_at"`
	IsDeleted       bool             `json:"is_deleted"`
	ViewCount       int              `json:"view_count"`
	AverageRating   float64          `json:"average_rating"`
	ReviewIds       []string         `json:"review_ids"`
}

type NutritionalInfo struct {
	Calories  int `json:"calories"`
	Protein   int `json:"protein"`
	Carbs     int `json:"carbs"`
	Fat       int `json:"fat"`
	DeletedAt *time.Time
}

type IItemRepository interface {
	GetItemByID(ctx context.Context, id string) (*Item, error)
	CreateItem(ctx context.Context, item *Item) error
	UpdateItem(ctx context.Context, id string, item *Item) error
	DeleteItem(ctx context.Context, id string) error
	AddReview(ctx context.Context, itemID, reviewID string) error
	GetItems(ctx context.Context, menuSlug string) ([]Item, error)
}

type IItemUseCase interface {
	CreateItem(item *Item) error
	UpdateItem(id string, item *Item) error
	GetItems(menuSlug string) ([]Item, error)
	GetItemByID(id string) (*Item, error)
	AddReview(itemID, reviewID string) error
	DeleteItem(id string) error
}

File: dinq-mobile/dinq/lib/features/dinq/restaurant_management/domain/entities/item.dart
import 'package:equatable/equatable.dart';

class Item extends Equatable {
  final String id;
  final String name;
  final String nameAm;
  final String slug;
  final String categoryId;
  final String? description;
  final List<String>? image;
  final int price;
  final String currency;
  final List<String>? allergies;
  final List<dynamic>? userImages;
  final int? calories;
  final List<String>? ingredients;
  final List<String>? ingredientsAm;
  final int? preparationTime;
  final String? howToEat;
  final String? howToEatAm;
  final int viewCount;
  final double averageRating;
  final List<String> reviewIds;
  final String? protein;
  final String? carbs;
  final String? fat;

  const Item({
    required this.id,
    required this.name,
    required this.nameAm,
    required this.slug,
    required this.categoryId,
    this.description,
    this.image,
    required this.price,
    required this.currency,
    this.allergies,
    this.userImages,
    this.calories,
    this.ingredients,
    this.ingredientsAm,
    this.preparationTime,
    this.howToEat,
    this.howToEatAm,
    this.protein,
    this.carbs,
    this.fat,
    required this.viewCount,
    required this.averageRating,
    required this.reviewIds,
  });

  @override
  List<Object?> get props {
    return [id];
  }
}
